mypy/subtypes.py

```diff
@@ -498,6 +498,19 @@ def visit_overloaded(self, left: Overloaded) -> bool:
             return False
 
     def visit_union_type(self, left: UnionType) -> bool:
+        if isinstance(self.right, Instance):
+            literal_types = set()  # type: Set[Instance]
+            # avoid redundant check for union of literals
+            for item in left.items:
+                if mypy.typeops.is_simple_literal(item):
+                    assert isinstance(item, LiteralType)  # type: ignore[misc]
```

Comment:
It might be a good idea to declare `item = get_proper_type(item)`

Reply:
or I suppose I might be able to use the new TypeGuard feature here?

Reply:
Yes 🙂 

But, I am not sure that `mypyc` supports it.

Reply:
black is compiled with mypyc and you added TypeGuard to it @sobolevn so it's at least worth a shot imo :slightly_smiling_face:

Reply:
well OK it turns out I misremembered sobolevn for someone else, but anyway the first half of my still stands 😅